src/com/google/javascript/jscomp/AbstractCommandLineRunner.java

```diff
@@ -368,7 +368,8 @@ protected void setRunOptions(CompilerOptions options)
 
     if (config.createSourceMap.length() > 0) {
       options.sourceMapOutputPath = config.createSourceMap;
-    } else if (config.useJsonStreams) {
+    } else if (config.jsonStreamMode == CompilerOptions.JsonStreamMode.OUT ||
+        config.jsonStreamMode == CompilerOptions.JsonStreamMode.BOTH) {
```

Comment:
The out || both check is the most common. You can define a private helper method in this file:

    private boolean isOutputInJson() {
      return config.jsonStreamMode == CompilerOptions.JsonStreamMode.OUT
            || config.jsonStreamMode == CompilerOptions.JsonStreamMode.BOTH;
    }